examples/architecture/JSONCtrProperties.wyt

```diff
@@ -0,0 +1,81 @@
+					val CSIface_provides_port = "module def "+ connectorIface + "_provides_port(port : " + connectorIface + ", tcpServer : io.ServerSocket, java : Java)\nimport thread.thread\nimport thread.Thread\nimport io.io\nimport io.Socket\nimport io.DataInputStream\nimport io.DataOutputStream\nimport json\nval thread = thread(java)\nval io = io(java)\nval json = json(java)\ndef run() : Unit\n\tval socket : Socket = tcpServer.accept()\n\tval outData : DataOutputStream = io.makeDataOutputStream(socket.getOutputStream())\n\tval inData : DataInputStream = io.makeDataInputStream(socket.getInputStream())\n\tval key : json.JSON = json.stringToJSON(inData.readUTF())\n\toutData.writeUTF(port.getVal(json.deserializeFromJSON(key)))\n\trun()\nval runningThread : Thread = thread.make(() => run())\ndef getThread() : Thread\n\trunningThread\n"
+					
+					val requiresAST = ast.parseGeneratedModule(CSIface_requires_port)
+					val providesAST = ast.parseGeneratedModule(CSIface_provides_port)
+					
+					portList.append(requiresAST)
+					portList.append(providesAST)
+					portList
+				else
+					list.make[ast.AST]()
+
+		def generateConnectorInit(portInstances : list.List[ast.AST], compInstances : list.List[archast.ComponentDecl]) : list.List[ast.AST]
+			val astList = list.make[ast.AST]()
+			// in the future these should be retrieved from a deployment AST
+			val hostIP = "\"localhost\""
+			val hostport = "40000"
+			
+			val client = "require stdout\nrequire java\nimport Client\nimport CSIface_requires_port\nimport CSIface\nimport JSONCtrProperties\nval jsonCtrProps : JSONCtrProperties = new\n\tval host : String = " + hostIP + "\n\tval prt : Int = " + hostport + "\nval client_getInfo: CSIface = CSIface_requires_port(jsonCtrProps, java, stdout)\nval client = Client(client_getInfo)\n"
+			val server = "require java\nimport io.io\nimport CSIface_provides_port\nimport Server\nval io = io(java)\nval server = Server()\nval server_sendInfo = CSIface_provides_port(server.sendInfo, io.makeServerSocket("+hostport+"), java)\nval serverThread = server_sendInfo.getThread()\nserverThread.start()\n"
+
+			val clientAST : ast.AST = ast.parseExpressionNoContext(client)
+			val serverAST : ast.AST = ast.parseExpressionNoContext(server)
+			astList.append(serverAST)
+			astList.append(clientAST)
+			astList
+
```

Comment:
A known issue here is that the generate* functions are hardcoded to generate functions from the CSIface interface of our example. The fix is to pass a context into which the interface type has been loaded to the generate* functions and use that to generate the appropriate functions. I'll fix this after the pull request is accepted.